### PR TITLE
Adding consent Receipt Rest API implementation. 

### PR DESCRIPTION
**Pull request by IsuraD**
[1] https://app.swaggerhub.com/apis/consent/Approval-Consent/1.0.0

> User Story 1 - Store User Consents

Scenario - Storing consents given by users
Background - Identity Server should have an API which supports storing consent information given by the users.
User Story - Store User Consents
User (Persona) - API Consumer

> User Story 2 - Retrieve A User Consent

Scenario - Retrieve a specific consent from consent ID
Background - Identity Server should have an API which supports retrieving a consent entry of a user, given the consent ID.
User Story - Retrieve A User Consent
User (Persona) - API Consumer

> User Story 3 - List User Consents

Scenario - List consents
Background - Identity Server should have an API which supports retrieving consent information of the users.
User Story - List User Consents
User (Persona) - API Consumer

> User Story 4 - Search User Consents

Scenario - Search consents
Background - Identity Server should have an API which supports searching consent information of the users.
User Story - Search User Consents
User (Persona) - API Consumer

> User Story 5 - Revoke User Consents

Scenario - Revoke consents
Background - Identity Server should have an API which supports revoke consent information of the users.
User Story - Revoke User Consents
User (Persona) - API Consumer

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/connector/PIIController.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This in
```
Comment (omindu) [inline components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/connector/PIIController.java]: Incomplete comment

File: components/org.wso2.carbon.consent.mgt.core/pom.xml [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+            <artifactId>org.wso2.carbon.logging</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.felix</groupId>
+            <artifactId>org.apache.felix.scr.ds-annotations</artifactId>
+            <version>1.2.4</version>
```
Comment (omindu) [inline components/org.wso2.carbon.consent.mgt.core/pom.xml]: Move version to parent pom

Comment (IsuraD) [reply components/org.wso2.carbon.consent.mgt.core/pom.xml]: Fixed.

File: components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/connector/PIIController.java (hunk 1 above)
Comment (IsuraD) [reply components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/connector/PIIController.java]: Fixed.

File: components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/dao/PurposeCategoryDAO.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.consent.mgt.core.dao;
+
+import org.wso2.carbon.consent.mgt.core.exception.ConsentManagementException;
+import org.wso2.carbon.consent.mgt.core.model.PurposeCategory;
+
+import java.util.List;
+
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/dao/PurposeCategoryDAO.java]: Shall we add some class level comments

File: components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/dao/PurposeDAO.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.consent.mgt.core.dao;
+
+import org.wso2.carbon.consent.mgt.core.exception.ConsentManagementException;
+import org.wso2.carbon.consent.mgt.core.model.Purpose;
+
+import java.util.List;
+
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/dao/PurposeDAO.java]: Shall we add some class level comments ?

File: components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/dao/ReceiptDAO.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.consent.mgt.core.dao;
+
+import org.wso2.carbon.consent.mgt.core.exception.ConsentManagementException;
+import org.wso2.carbon.consent.mgt.core.model.Receipt;
+import org.wso2.carbon.consent.mgt.core.model.ReceiptInput;
+
+public interface ReceiptDAO {
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/dao/ReceiptDAO.java]: Shall we add some class level comments ?

**Pull request merged**